src/material/core/typography/_all-typography.scss

```diff
@@ -33,11 +35,82 @@
 @use '../../toolbar/toolbar-theme';
 @use '../../tooltip/tooltip-theme';
 @use '../../snack-bar/snack-bar-theme';
-@use '../option/option-theme';
-@use '../option/optgroup-theme';
 @use '../../form-field/form-field-theme';
 @use '../../tree/tree-theme';
 @use '../theming/theming';
+@use '../core-theme';
+@use '../mdc-helpers/mdc-helpers';
+
+// TODO(mmalerba): define-mdc-typography-config is defined here rather than in _typography.scss
+//   (where define-typography-config is defined) because putting it there would cause a circular
+//   dependency with mdc-helpers. We should refactor so these can live in the same place.
+
+/// Generates an Angular Material typography config based on values from the official Material
+/// Design spec implementation (MDC Web). All arguments are optional, but may be passed to override
+/// the default values. The `mat-typography-level` function can be used to generate a custom
+/// typography level map which can be passed to this function to override one of the default levels.
+///
+/// @param {String} $font-family The font family to use for levels where it is not explicitly
+///     specified.
+/// @param {Map} $headline-1 The font settings for the headline-1 font level.
+/// @param {Map} $headline-2 The font settings for the headline-2 font level.
+/// @param {Map} $headline-3 The font settings for the headline-3 font level.
+/// @param {Map} $headline-4 The font settings for the headline-4 font level.
+/// @param {Map} $headline-5 The font settings for the headline-5 font level.
+/// @param {Map} $headline-6 The font settings for the headline-6 font level.
+/// @param {Map} $subtitle-1 The font settings for the subtitle-1 font level.
+/// @param {Map} $subtitle-2 The font settings for the subtitle-2 font level.
+/// @param {Map} $body-1 The font settings for the body-1 font level.
+/// @param {Map} $body-2 The font settings for the body-2 font level.
+/// @param {Map} $caption The font settings for the caption font level.
+/// @param {Map} $button The font settings for the button font level.
+/// @param {Map} $overline The font settings for the overline font level.
+/// @return {Map} A map containing font settings for each of the levels in the Material Design spec.
+@function define-mdc-typography-config(
```

Comment:
Should we merge this into `define-typography-config` instead?

Reply:
Oh thanks for pointing that out, I wanted to have a discussion about this. I'll post on our team chat

Reply:
Based on our discussion, I'll do a followup PR where I rename this and add in a way for people who need `px` units to more easily switch to this function. I added a TODO for now